src/schematools/loaders.py

```diff
@@ -0,0 +1,35 @@
+import importlib
+import os
+from typing import TYPE_CHECKING, Callable, Optional
+
+from schematools.types import DatasetSchema
+from schematools.utils import dataset_schema_from_file, dataset_schema_from_url
+
+
+def get_loader(reference: str) -> Callable:
+    """Gets a loader callable from a string reference."""
+    module_name, separator, callable_name = reference.partition(":")
+    if not separator:
+        raise ValueError("Provide a fully qualified reference `package.module:callable`")
+    obj = importlib.import_module(module_name)
+    for attr in callable_name.split("."):
+        obj = getattr(obj, attr)
+    return obj
+
+
+def schema_file_loader(dataset_id: str) -> Optional[DatasetSchema]:
+    """Load a datasetschema from the filesystem."""
+    try:
+        schemas_path = os.environ["SCHEMA_PATH"]
```

Comment:
Could we consider using Pydantic for dealing with settings and environment variables: https://pydantic-docs.helpmanual.io/usage/settings/

Reply:
> `file:` URLs can be anything we want, because we interpret them :) They're awkward to use, though.
> 
> Even easier is to just treat the "URL" as a filename if it doesn't start with `http` or `https`:
> 
> ```python
> u = urlparse(location)
> if u.scheme in ("http", "https"):
>     # treat location as URL
> else:
>     # treat location as local path
> ```
> 
> (or just do `location.startswith("https://") or location.startswith("http://")`)

Makes it simpler and is easy to understand. And, that's what is also being done in the cli code.

Reply:
+1 for Pydantic. But, this should be used for all enviroment acces in the whole repo. Agreed?

Reply:
Agreed